pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -29,6 +31,31 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+const (
+	LeaseDurationInSeconds = 3600
+	LeaseHolderIdentity    = "node-maintenance"
+	LeaseNamespace         = "node-maintenance-operator"
+	RequeWaitTime          = 10 * time.Second // time to reqeueu draining if not completed.
+	LeasePaddingSeconds    = 30
```

Comment:
Arguably the length of the maintenance window is determined by the lifespan of the CR, so perhaps we don't need `LeasePaddingSeconds`, and `LeaseDurationInSeconds` instead defines how long we ask for at a time.

Reply:
removed LeasePaddingSeconds. Now I don't see any  lifespan/time period specified in the CRD, NodeMaintenanceSpec doesn't have one.

Reply:
lifespan == the time between when the CR is created and destroyed. not an actual field

Reply:
`LeasePaddingSeconds` still seems to exist.

The NMO should ask for the lease with a duration of `LeaseDurationInSeconds` for as long as the CR exists.
If the `Reconcile()` loop detects that the lease is about to expire, it should ask for another interval of `LeaseDurationInSeconds`.

Reply:
yes, I added that in updateLease - if the lease is already owned by us and is to expire within twice the length of the drainer timeout then we are extending the lease (without setting acquireTime and without incrementing leasetransitions)

Reply:
`LeasePaddingSeconds` (not the same thing as `LeaseDurationInSeconds`) is not used anywhere. Please remove it.

Reply:
Not done

Reply:
removed in f6e44b2a6f4437cbca3f3f50073f76f863c96c31